service/src/event/service.rs

```diff
@@ -71,35 +67,60 @@ impl CeramicEventService {
             .await?;
         Ok(())
     }
-
     /// This function is used to parse the event from the carfile and return the insertable event and the previous cid pointer.
     /// Probably belongs in the event crate.
-    pub(crate) async fn parse_event_carfile(
-        event_cid: cid::Cid,
+    pub(crate) async fn parse_event_carfile_order_key(
+        event_id: EventId,
         carfile: &[u8],
-    ) -> Result<(EventInsertableBody, Option<DeliverableMetadata>)> {
-        let insertable = EventInsertableBody::try_from_carfile(event_cid, carfile).await?;
-        let ev_block = insertable.block_for_cid(&insertable.cid)?;
+    ) -> Result<(EventInsertable, EventHeader)> {
+        let mut insertable = EventInsertable::try_from_carfile(event_id, carfile).await?;
+
+        let header = Self::parse_event_body(&mut insertable.body).await?;
+        Ok((insertable, header))
+    }
 
-        trace!(count=%insertable.blocks.len(), cid=%event_cid, "parsing event blocks");
+    pub(crate) async fn parse_event_carfile_cid(
+        cid: ceramic_core::Cid,
+        carfile: &[u8],
+    ) -> Result<InsertableBodyWithHeader> {
+        let mut body = EventInsertableBody::try_from_carfile(cid, carfile).await?;
+
+        let header = Self::parse_event_body(&mut body).await?;
+        Ok(InsertableBodyWithHeader { body, header })
+    }
+
+    pub(crate) async fn parse_event_body(body: &mut EventInsertableBody) -> Result<EventHeader> {
+        let cid = body.cid(); // purely for convenience writing out the match
+        let ev_block = body.block_for_cid(&cid)?;
+
+        trace!(count=%body.blocks().len(), %cid, "parsing event blocks");
         let event_ipld: unvalidated::RawEvent<Ipld> =
             serde_ipld_dagcbor::from_slice(&ev_block.bytes).map_err(|e| {
                 Error::new_invalid_arg(
                     anyhow::anyhow!(e).context("event block is not valid event format"),
                 )
             })?;
 
-        let maybe_init_prev = match event_ipld {
-            unvalidated::RawEvent::Time(t) => Some((t.id(), t.prev())),
+        let (deliverable, header) = match event_ipld {
```

Comment:
can we re-use `Event::decode_car` (https://github.com/ceramicnetwork/rust-ceramic/blob/5d40086ecb30fe0fb7e1a490ead72c958012c5cd/event/src/unvalidated/event.rs#L75) to simplify the logic here and reduce code duplication?